contracts/protocols/mainnet/morpho/helpers.sol

```diff
@@ -0,0 +1,369 @@
+                .getCurrentSupplyBalanceInOf(poolTokenAddress, user);
+            userMarketData_.borrowRate = aavelens.getCurrentUserBorrowRatePerYear(poolTokenAddress, user);
+            userMarketData_.supplyRate = aavelens.getCurrentUserSupplyRatePerYear(poolTokenAddress, user);
+        } else {
+            (userMarketData_.p2pBorrows, userMarketData_.poolBorrows, userMarketData_.totalBorrows) = compLens
+                .getCurrentBorrowBalanceInOf(poolTokenAddress, user);
+            (userMarketData_.p2pSupplies, userMarketData_.poolSupplies, userMarketData_.totalSupplies) = compLens
+                .getCurrentSupplyBalanceInOf(poolTokenAddress, user);
+            userMarketData_.borrowRate = compLens.getCurrentUserBorrowRatePerBlock(poolTokenAddress, user);
+            userMarketData_.supplyRate = compLens.getCurrentUserSupplyRatePerBlock(poolTokenAddress, user);
+        }
+        (userMarketData_.maxWithdrawable, userMarketData_.maxBorrowable) = lens_.getUserMaxCapacitiesForAsset(
+            user,
+            poolTokenAddress
+        );
+    }
+
+    function getUserMarkets(address user, Underlying pool) internal view returns (address[] memory userMarkets_) {
+        ILens lens_ = pool == Underlying.AAVEV2 ? ILens(getAaveLens()) : ILens(getCompoundLens());
+        userMarkets_ = lens_.getEnteredMarkets(user);
+    }
+
+    function getUserData(
+        address user,
+        address[] memory poolTokenAddresses,
+        Underlying pool
+    ) internal view returns (UserData memory userData_) {
+        uint256 length_ = poolTokenAddresses.length;
+
+        UserMarketData[] memory marketData_ = new UserMarketData[](length_);
+        for (uint256 i = 0; i < length_; i++) {
+            marketData_[i] = getUserMarketData(user, poolTokenAddresses[i], pool);
+        }
+
+        userData_.marketData = marketData_;
+        // uint256 unclaimedRewards;
+        address[] memory userMarkets_ = getUserMarkets(user, pool);
+
+        if (pool == Underlying.AAVEV2) {
+            userData_.healthFactor = aavelens.getUserHealthFactor(user);
+            (
+                userData_.collateralValue,
+                userData_.maxDebtValue,
+                userData_.liquidationThreshold,
+                userData_.debtValue
+            ) = aavelens.getUserBalanceStates(user);
+            userData_.isLiquidatable = aavelens.isLiquidatable(user);
+        } else {
+            userData_.healthFactor = compLens.getUserHealthFactor(user, userMarkets_);
+            (userData_.collateralValue, userData_.debtValue, userData_.maxDebtValue) = compLens.getUserBalanceStates(
+                user,
+                userMarkets_
+            );
+            userData_.isLiquidatable = compLens.isLiquidatable(user, userMarkets_);
+            userData_.unclaimedRewards = compLens.getUserUnclaimedRewards(userMarkets_, user);
+        }
+    }
+
+    function getMorphoData() internal view returns (MorphoData memory morphoData_) {
```

Comment:
@Richa-iitr if Morpho adds more protocols, we need to update the resolver everytime?

Reply:
Yes, this needs to be done as they have separate contracts for separate protocols.